azure-cs-botservice/Program.cs

```diff
@@ -0,0 +1,127 @@
+                },
+            });
+
+            var container = new Container("zips", new ContainerArgs
+            {
+                StorageAccountName = storageAccount.Name,
+                ContainerAccessType = "private",
+            });
+
+            var blob = new ZipBlob("zip", new ZipBlobArgs
+            {
+                StorageAccountName = storageAccount.Name,
+                StorageContainerName = container.Name,
+                Type = "block",
+                Content = new FileArchive("bot/bot.zip")
+            });
+
+            var codeBlobUrl = SharedAccessSignature.SignedBlobReadUrl(blob, storageAccount);
+
+            var appInsights = new Pulumi.Azure.AppInsights.Insights("ai", new Pulumi.Azure.AppInsights.InsightsArgs
+            {
+                ApplicationType = "web",
+                ResourceGroupName = resourceGroup.Name
+            });
+
+            var appInsightApiKey = new Pulumi.Azure.AppInsights.ApiKey("ai", new Pulumi.Azure.AppInsights.ApiKeyArgs
+            {
+                ApplicationInsightsId = appInsights.Id,
+                ReadPermissions = "api",
+            });
+
+            var luis = new Pulumi.Azure.Cognitive.Account("cs", new Pulumi.Azure.Cognitive.AccountArgs
+            {
+                Kind = "CognitiveServices", // includes LUIS
+                ResourceGroupName = resourceGroup.Name,
+                Sku = new Pulumi.Azure.Cognitive.Inputs.AccountSkuArgs() { Name = "S0", Tier = "Standard" }
+            });
+
+            var msa = new Application("msapp", new ApplicationArgs
+            {
+                Oauth2AllowImplicitFlow = false,
+                AvailableToOtherTenants = true,
+                PublicClient = true
+            });
+
+            var msaSecret = new ApplicationPassword("msapp", new ApplicationPasswordArgs
+            {
+                ApplicationObjectId = msa.ObjectId,
+                EndDateRelative = "8640h",
+                Value = botSecret
+            });
+
+            var app = new AppService("app", new AppServiceArgs
+            {
+                ResourceGroupName = resourceGroup.Name,
+                AppServicePlanId = appServicePlan.Id,
+                AppSettings =
+                {
+                    { "WEBSITE_RUN_FROM_PACKAGE", codeBlobUrl },
+                    { "MicrosoftAppId", msa.ApplicationId },
+                    { "MicrosoftAppPassword", msaSecret.Value },
+                    { "LuisApiKey", luis.PrimaryAccessKey },
+                },
+                HttpsOnly = true
+            });
+
+            var bot = new Pulumi.Azure.Bot.WebApp(botName, new Pulumi.Azure.Bot.WebAppArgs
+            {
+                DisplayName = botName,
+                MicrosoftAppId = msa.ApplicationId,
+                ResourceGroupName = resourceGroup.Name,
+                Sku = "F0",
+                Location = "global",
+                Endpoint = app.DefaultSiteHostname.Apply(a => $"https://{a}/api/messages"),
```

Comment:
I prefer `Output.Format` as you do below.

Reply:
Done